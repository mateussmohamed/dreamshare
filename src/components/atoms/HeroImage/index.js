import styled from 'styled-components'
import { position, size } from 'polished'

const getBackGround = src => `
  linear-gradient(to bottom, rgba(0, 0, 0, 0.3), rgba(0, 0, 0, 0.3)),
  url(${src})
`

const HeroImage = styled.div`
  ${size('100%', '100%')};
  ${position('absolute', '0')};

  background-size: cover;
  background-position: center center;
  background-attachment: fixed;
  background-repeat: no-repeat;
  z-index: 0;

  @media only screen and (min-width: 320px) {
    background-image: ${() => getBackGround('/assets/images/hero-mobile.jpg')};
  }

  @media only screen and (min-width: 768px) {
    background-image: ${() => getBackGround('/assets/images/hero-tablet.jpg')};
  }

  @media only screen and (min-width: 768px) {
    background-image: ${() => getBackGround('/assets/images/hero.jpg')};
  }
`

export default HeroImage
