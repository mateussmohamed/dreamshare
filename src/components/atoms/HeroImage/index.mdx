---
name: HeroImage
menu: Atoms
---

import { Playground, PropsTable } from 'docz'
import HeroImage from './index'

# HeroImage
<PropsTable of={HeroImage} />


<Playground style={{ padding: 0 }}>
  <div style={{ width: '100%', height: '1080px' }}>
    <HeroImage  />
  </div>
</Playground>
