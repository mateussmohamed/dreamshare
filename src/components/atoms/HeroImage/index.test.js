import React from 'react'
import { shallow } from 'enzyme'
import 'jest-styled-components'

import HeroImage from './index'

describe('[<HeroImage />]', () => {
  const wrap = (props = {}) => shallow(<HeroImage {...props} />).dive()

  test('matches to snapshot', () => {
    const wrapper = wrap({ src: 'assets/hero.jpg' })

    expect(wrapper).toMatchSnapshot()
  })
})
