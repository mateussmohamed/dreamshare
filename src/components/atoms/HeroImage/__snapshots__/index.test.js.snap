// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[<HeroImage />] matches to snapshot 1`] = `
<StyledComponent
  forwardedClass={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": undefined,
      "componentStyle": ComponentStyle {
        "componentId": "HeroImage-sc-1g3c219-0",
        "isStatic": false,
        "rules": Array [
          "height: 100%; width: 100%;",
          ";",
          "position: absolute; top: 0; right: 0; bottom: 0; left: 0;",
          ";background-size:cover;background-position:center center;background-attachment:fixed;background-repeat:no-repeat;z-index:0;@media only screen and (min-width:320px){background-image:",
          [Function],
          ";}@media only screen and (min-width:768px){background-image:",
          [Function],
          ";}@media only screen and (min-width:768px){background-image:",
          [Function],
          ";}",
        ],
      },
      "displayName": "HeroImage",
      "render": [Function],
      "styledComponentId": "HeroImage-sc-1g3c219-0",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;
